data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostAttesterSlashing.java

```diff
@@ -69,25 +89,37 @@ public PostAttesterSlashing(final NodeDataProvider provider, final JsonProvider
         @OpenApiResponse(status = RES_INTERNAL_ERROR),
       })
   @Override
-  public void handle(final Context ctx) throws Exception {
-    try {
-      final AttesterSlashing attesterSlashing =
-          parseRequestBody(ctx.body(), AttesterSlashing.class);
-      InternalValidationResult result =
-          nodeDataProvider.postAttesterSlashing(attesterSlashing).join();
-      if (result.code().equals(ValidationResultCode.IGNORE)
-          || result.code().equals(ValidationResultCode.REJECT)) {
-        ctx.status(SC_BAD_REQUEST);
-        ctx.json(
-            BadRequest.badRequest(
-                jsonProvider,
-                "Invalid attester slashing, it will never pass validation so it's rejected."));
-      } else {
-        ctx.status(SC_OK);
-      }
-    } catch (final IllegalArgumentException | JsonMappingException e) {
-      ctx.json(BadRequest.badRequest(jsonProvider, e.getMessage()));
-      ctx.status(SC_BAD_REQUEST);
-    }
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final AttesterSlashing attesterSlashing = request.getRequestBody();
+    final SafeFuture<InternalValidationResult> future =
+        nodeDataProvider.postAttesterSlashing(attesterSlashing);
+
+    request.respondAsync(
+        future.thenApply(
+            internalValidationResult -> {
+              if (internalValidationResult.code().equals(ValidationResultCode.IGNORE)
+                  || internalValidationResult.code().equals(ValidationResultCode.REJECT)) {
+                return AsyncApiResponse.respondWithError(
+                    SC_BAD_REQUEST,
+                    "Invalid attester slashing, it will never pass validation so it's rejected.");
```

Comment:
one potential thing we could improve to be more helpful... the internalValidationResult has an optional description, so we could return that if its set...
```
return AsyncApiResponse.respondWithError(SC_BAD_REQUEST,
   internalValidationResult.getDescription()
      .orElse("Invalid attester slashing, it will never pass validation so it's rejected."));
```